src/main/kotlin/com/hibob/academy/types/StoreService.kt

```diff
@@ -0,0 +1,80 @@
+package com.hibob.academy.types
+
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+import Cart
+import Payment
+import Check
+import Statuses
+
+const val ZERO_TOTAL_COSTS_ON_FAILED_BUY = 0.0
+const val VISA_LEGIT_NUM_LENGTH = 10
+
+fun fail(message: String) : Nothing?{
+    throw IllegalStateException(message)
+}
+
+@Component
+class StoreService {
+
+    fun pay(cart: List<Cart>, payment: Payment): Map<String, Check> {
+        val result = mutableMapOf<String, Check>()
+
+        for (c in cart) {
+            val check = checkout(c, payment)
+
+            result[c.clientId] = check
+        }
+        return result
+    }
+
+    private fun isCustomTrue(custom: Any): Boolean {
+
+        return when(custom) {
+            is Boolean -> custom
+            else -> false
+        }
+    }
+
+    private fun checkout(cart: Cart, payment: Payment): Check {
+        var totalPricesOfLegitProducts = 0.0
+
+        cart.products.forEach { product ->
+            if (isCustomTrue(product.custom)) {
+                totalPricesOfLegitProducts += product.price
+            }
+        }
+
+        val status: Statuses = when(payment) {
+            is Payment.Cash -> {
+                fail("payment is Cash")
```

Comment:
after you throw an error, you don't need to return value, you use nothing function